tests/api/utils/test_scheduler.py

```diff
@@ -1107,3 +1122,21 @@ def _create_mlrun_function_and_matching_scheduled_object(
         }
     }
     return scheduled_object
+
+
+def _get_start_and_end_time_for_scheduled_trigger(
+    number_of_jobs: int, seconds_interval: int
+):
+    """
+    If microsecond is 0, then the scheduler will be able to fit one more run, because the scheduler
+    starts at the start of every second(when microsecond == 0) so if for example we want to run between
+    12:08:06.100000 until 12:08:07.10000 the scheduler will run a job at 12:08:07.00000
+    (because the start request is already past the time cron starts to run), but if microsecond is 0
+    12:08:06.000000 until 12:08:07:000000 the scheduler will run a job also at 06 and 07 second.
```

Comment:
I tried to construct it differently - starting from explaining how it works, going through examples, and then explaining the implications, our solution, and why we chose it.
You don't have to agree this is better, if you do prefer going with your "flow of explanations" - just go over the english to make sure it's correct
```suggestion
    The scheduler executes the job on round seconds (when microsecond == 0)
    Therefore if the start time will be a round second - let's say 12:08:06.000000 and the end time 12:08:07.000000 it means two executions will happen - at 06 and 07 second.
    This is obviously very rare (since the times are based on datetime.now()) - usually the start time will be something like 12:08:06.100000 then the end time will be 12:08:07.10000 - meaning there will be only one execution on the 07 second.
    So instead of conditioning every assertion we're doing on whether the start time was a round second, we simply make sure it's not a round second
```